pages/agent/v3.md

```diff
@@ -165,6 +130,57 @@ The job API is unavailable on agents running versions of Windows before build 17
 >🛠 Experimental feature
 > To use the job API, set <code>experiment="job-api"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a>.
 
+### Agent API
+
+Like the Job API experiment, this exposes a (separate) local API for interacting with the agent process.
+The Agent API offers these endpoints:
+
+- `GET /api/leader/v0/ping` - Returns a JSON object with the current time (useful for testing the agent is alive).
+- `GET /api/leader/v0/lock?key=<key>` - Returns a JSON object containing the current state of a lock.
+- `PATCH /api/leader/v0/lock?key=<key>` - Accepts a JSON object with old and new states for a lock. The lock is then updated atomically, and a JSON object describing whether the operation proceeded is returned.
+
+The API is exposed using a Unix Domain Socket. Unlike the `job-api`, the path to the socket is not available through a environment variable—rather, there is a single (configurable) path on the system.
+
+>🛠 Experimental feature
+> To use the agent API, set <code>experiment="agent-api"</code> in your <a href="/docs/agent/v3/configuration#experiment"> agent configuration</a>.
+
+
+## Promoted experiments
+
+The following features started as experiments before being promoted to fully supported features.
+
+### Flock file locks
+
+Changes the file lock implementation from `github.com/nightlyone/lockfile` to `github.com/gofrs/flock` to address an issue where file locks are never released by agents that don't shut down cleanly. The new file locks are implemented at the kernel level, and are aware of when their parent process dies.
+
+Promoted in [v3.48.0](https://github.com/buildkite/agent/releases/tag/v3.48.0). It's the default behavior, so there's no configuration required to use it. Because the old and new lock systems do not interact, we *strongly* recommend not running different versions of the agent on the same host.
+
+### ANSI timestamps
+
+Outputs inline ANSI timestamps for each line of log output which enables toggle-able timestamps in the Buildkite UI.
```

Comment:
Minor nitpicks:

- Missing comma with "which" clause.
- "toggle-able" – I'm not sure. 😆
- Buildkite UI -> Buildkite dashboard.

```suggestion
Outputs inline ANSI timestamps for each line of log output, enabling timestamps you can toggle in the Buildkite dashboard.
```